main.go

```diff
@@ -72,11 +77,13 @@ func run(ec2Service ec2iface.EC2API, autoScalingService autoscalingiface.AutoSca
 		autoScalingGroups, err = cloud.DescribeAutoScalingGroupsByNames(autoScalingService, cfg.AutoScalingGroupNames)
 	}
 	if err != nil {
+		metrics.Server.Errors.Inc()
 		return errors.New("unable to describe AutoScalingGroups: " + err.Error())
 	}
 	if cfg.Debug {
 		log.Println("Described AutoScalingGroups successfully")
 	}
+	metrics.Server.NodeGroups.WithLabelValues().Set(float64(len(autoScalingGroups)))
 	return HandleRollingUpgrade(kubernetesClient, ec2Service, autoScalingService, autoScalingGroups)
```

Comment:
Can you move `metrics.Server.NodeGroups.WithLabelValues().Set(float64(len(autoScalingGroups)))` inside `HandleRollingUpgrade`?